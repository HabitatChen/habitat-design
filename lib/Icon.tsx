import React from 'react'
import './importAllIcons'

/**
 * 1. 添加一个loader ==> 转换svg源代码
 * 2. 把所有svg文件放到 icons 文件中
 * 3. 在页面上使用 svg 然后使用 use 标签添加 xlink-href 属性
 */

interface IconProps {
    name: string
}

const Icon: React.FunctionComponent<IconProps> = (props: any) => {
    return (
        <span>
            <svg>
                <use xlinkHref={`#${ props.name }`} />
            </svg>
        </span>
    )
}

export default Icon