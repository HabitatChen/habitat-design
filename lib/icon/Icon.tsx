import React from 'react'
import './importAllIcons.js'
import './icon.scss'
// @ts-ignore
import classes from '../helppers/classnames.tsx'

/**
 * 1. 添加一个loader ==> 转换svg源代码
 * 2. 把所有svg文件放到 icons 文件中
 * 3. 在页面上使用 svg 然后使用 use 标签添加 xlink-href 属性
 */

interface IconProps extends React.SVGAttributes<SVGElement> {
    name: string
}

const Icon: React.FunctionComponent<IconProps> = ({ className, name, ...restProps }: any) => {
    return (
        <svg
            className={classes('icon', className)}
            { ...restProps }
        >
            <use xlinkHref={`#${ name }`} />
        </svg>
    )
}

export default Icon