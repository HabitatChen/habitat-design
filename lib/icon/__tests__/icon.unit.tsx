import renderer from 'react-test-renderer'
import * as React from 'react'
// @ts-ignore
import Icon from '../Icon.tsx'
// @ts-ignore
import { mount } from 'enzyme'

describe('icon', () => {
    it('是个div', () => {
        // 渲染一个虚拟ODM 就是一个对象
        let json = renderer.create(<Icon name={'alipay'} />).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('测试点击事件', () => {
        // 这种方式太low了
        //  let n = 1
        //  const fn = () => { n = 2 }
        const fn = jest.fn()
        const component = mount(<Icon name={'alipay'} onClick={fn}/>)
        // 模拟点击事件
        component.find('svg').simulate('click')
        expect(fn).toBeCalled()
    })
})