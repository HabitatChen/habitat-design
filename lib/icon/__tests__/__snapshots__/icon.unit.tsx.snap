// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 是个div 1`] = `
<svg
  className="icon"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
